python/tk_houdini_renderman/handler.py

```diff
@@ -77,65 +65,61 @@ def submit_to_farm(self, node, network):
-            self.app, node, file_name, "50", framerange, network=network
+            self.app, node, file_name, 50, framerange, render_paths, network=network
         )
         submission.show()
 
     def execute_render(self, node, network):
-        # Validate node
-        if not self.__validate_node(node, network):
-            return
-
-        # Prepare node and set all parameters
-        render_path = self.__calculate_path(node, network)
-        self.__apply_path(node, render_path, network)
-        self.__create_directory(render_path)
-
-        # Set filters
-        if network == "lop":
-            filters = self.__check_lop_filters(node)
-            self.__set_lop_filter_filename(node, filters)
-
-        if network == "rop":
-            filters = self.__check_rop_filters(node)
-            self.__set_rop_filter_filename(filters, render_node=node)
+        # Create directories
+        render_paths = node.type().hdaModule().get_render_paths()
+        for p in render_paths:
+            self.__create_directory(p)
+
+        # # Set filters
+        # if network == "lop":
+        #     filters = self.__check_lop_filters(node)
+        #     self.__set_lop_filter_filename(node, filters)
+        #
+        # if network == "rop":
+        #     filters = self.__check_rop_filters(node)
+        #     self.__set_rop_filter_filename(filters, render_node=node)
 
         # Execute rendering
         if network == "lop":
             node.node("rop_usdrender").parm("execute").pressButton()
         else:
-            node.node("ris1").parm("execute").pressButton()
+            node.node("denoise" if node.evalParm("denoise") else "render").parm(
+                "execute"
+            ).pressButton()
 
     def copy_to_clipboard(self, node, network=None):
         """Function to copy the path directly to the clipboard,
         currently only Windows is supported
 
         Args:
-            node (attribute): node to get clipboard from
+            :param node: node to get clipboard from
+            :param network:
         """
 
         # Function to copy the path directly to the clipboard,
         # currently only Windows is supported
         if platform.system() == "Windows":
-
             if network == "rop":
                 parameter = "ri_display_0"
             else:
                 parameter = "picture"
 
             render_path = node.parm(parameter).eval()
             render_path = os.path.dirname(render_path).replace("/", os.sep)
-            copy_to_clipboard = (
-                'echo|set /p="' + render_path.strip() + '"| clip'
-            )
+            copy_to_clipboard = 'echo|set /p="' + render_path.strip() + '"| clip'
             os.system(copy_to_clipboard)
         else:
             self.app.logger.debug(
                 "Currently copying to clipboard is only supported on Windows."
             )
 
     @staticmethod
-    def __validate_node(node, network):
+    def validate_node(node, network):
```

Comment:
Incorrect pep8 formatting, should look like:
```python
def validate_node(node, network):
        """This function will make sure all the parameters
        are filled in and setup correctly.
        First we'll check if there is a name"""
        render_name = node.parm("name").eval()
```

instead of
```python
def validate_node(node, network):
        #This function will make sure all the parameters
        #are filled in and setup correctly.
        #First we'll check if there is a name
        render_name = node.parm("name").eval()
```

Use # for inline comments

Reply:
Didn't change Gilles's old code where not nessecary but will fix/add the function descriptions where needed.